macros/utils/cross_db_utils/generate_elementary_profile_args.sql

```diff
@@ -0,0 +1,107 @@
+{% macro generate_elementary_profile_args(method=none, overwrite_values=none) %}
+  {#
+    Returns a list of parameters for elementary profile.
+    Each parameter consists of a dict consisting of: 
+      name
+      value
+      comment
+  #}
+  {% set elementary_database, elementary_schema = elementary.get_package_database_and_schema() %}
+  {% set parameters = adapter.dispatch('generate_elementary_profile_args', 'elementary')(method, elementary_database, elementary_schema) %}
+  {% if overwrite_values %}
+    {% for parameter in parameters %}
+      {% if parameter["name"] in overwrite_values %}
+        {% do parameter.update({"value": overwrite_values[parameter["name"]]}) %}
+      {% endif %}
+    {% endfor %}
+  {% endif %}
+  {% do return(parameters) %}
+{% endmacro %}
+
+{% macro _parameter(name, value, comment=none) %}
+  {% do return({"name": name, "value": value, "comment": comment}) %}
+{% endmacro %}
+
+{% macro snowflake__generate_elementary_profile_args(method, elementary_database, elementary_schema) %}
+  {% do return([
+    _parameter("type", target.type),
```

Comment:
Technically this could be overwritten by a user with the same macro because we don't prefix with `elementary.`, but it's extremely unlikely so I suppose that's negligible.